### PR TITLE
[Attestation Service] MessageBird provider check for KYCed US-based number

### PR DESCRIPTION
**Pull request by timmoreton**
### Description

Error on startup if MessageBird provider if:
* API key is incorrect
* No US numbers found that are enabled for SMS, and have passed KYC.

### Tested

Locally

### Backwards compatibility

Yes

### CODE REVIEW

File: packages/attestation-service/src/sms/messagebird.ts [hunk 1]
```diff
@@ -61,12 +64,36 @@ export class MessageBirdSmsProvider extends SmsProvider {
     return [bodyParser.urlencoded({ extended: false })]
   }
 
+  async getUSNumbers(): Promise<string[]> {
```
Comment (aslawson) [inline packages/attestation-service/src/sms/messagebird.ts:67]: didn't they mentioned this is the case for canadian numbers as well? or only US numbers which are required for sending to US/Canada?

File: packages/attestation-service/src/sms/messagebird.ts [hunk 2]
```diff
@@ -61,12 +64,36 @@ export class MessageBirdSmsProvider extends SmsProvider {
...
+        Authorization: `AccessKey ${this.apiKey}`,
+      },
+    })
+    if (!response.ok) {
+      throw new Error('Could not list numbers! ' + response.status)
+    }
+    const body = JSON.parse(await response.text())
+    return body.items
+      ? body.items
+          .filter((n: any) => n.country === 'US' && n.kycStatus === 'ok')
```
Comment (aslawson) [inline packages/attestation-service/src/sms/messagebird.ts:81]: oh nice.  interesting that the KYC specific to the number rather than the account.

**Review by aslawson — APPROVED** [review]
interesting -- i had thought we'd add this check when selecting provider to be the sender based on locale to send. 

it sounded like we only needed US number for sending to US/Canada, so if the validator has a non-US number, could they still send to other non-US/Canada numbers without the KYC? but if its to a US number, then we just remove messagebird from the list of possible providers for that request.

no problem if we'd rather not complicate it and just require all validators to have a KYCd US number in order to use message bird though 👍

**Comment by timmoreton** [comment]
> interesting -- i had thought we'd add this check when selecting provider to be the sender based on locale to send.

In that case, the check would make no difference vs current behaviour: the send would fail anyway, and the service would then immediately retry with the next provider.  

> it sounded like we only needed US number for sending to US/Canada, so if the validator has a non-US number, could they still send to other non-US/Canada numbers without the KYC? but if its to a US number, then we just remove messagebird from the list of possible providers for that request.

That's true (well, and some other countries) but if we are relying on MessageBird (e.g. as the only provider) I think we really want the Attestation Service not be able to start and force the validator do KYC before deploying it.  We don't want to be in the situation where it can't send to the US.   But we could consider changing the logic so it only aborts if no providers can be successfully configured...? 

> no problem if we'd rather not complicate it and just require all validators to have a KYCd US number in order to use message bird though 👍

**Comment by aslawson** [comment]
>> interesting -- i had thought we'd add this check when selecting provider to be the sender based on locale to send.

> In that case, the check would make no difference vs current behaviour: the send would fail anyway, and the service would then immediately retry with the next provider.

on first request, doesn't it generate a list of all possible providers? it check the country specific lists.  then that list is persisted in the db for subsequent requests.

```
    let providers: SmsProvider[] = countryCode
      ? smsProvidersFor(countryCode, phoneNumber, logger)
      : []
```
^ could filter here instead

but i'm fine if we want to enforce all validators just doing the KYC process with a US number to use MessageBird.

**Pull request merged**